cowprotocol/cow-amm
# Tell watchtower to retry if order size is too small

In the current setup, if `getTradeableOrder` is called and the traded amount is too low, then it reverts with `OrderNotValid("traded amount too small")`.
This has the drawback of telling the watchtower to stop creating further orders for the created conditional orders. From the watchtower logs:

```
Check conditional order result: ✋ DONT_TRY_AGAIN. Reason: ORDER_NOT_VALID: Invalid order: ORDER_NOT_VALID: traded amount too small
```

This is not the desired behavior, as we want to retry regularly after a fixed amount of time.

This PR fixes the issue by reverting with [the standard custom error](https://docs.cow.fi/cow-protocol/reference/contracts/periphery/composable-cow#custom-error-codes) that tells to retry at a future timestamp. I chose the time after which the order that would have been created would have expired.

### How to test

I deployed the contracts on Sepolia and tried them out at [this address](https://explorer.cow.fi/sepolia/address/0xe2f780347649d579452d6dc6960a197e6424580c).
The order was [created](https://sepolia.etherscan.io/tx/0x847b4f401cf101446ea0c93633b98383da8ca4df4fcec3b88e5817e790179e85) so that the first trade was too small, as seen in the logs:
```
Check conditional order result: 🕣 TRY_AT_EPOCH. Reason: POLL_TRY_AT_EPOCH: traded amount too small
```

By changing the reference price, you can see that the order is picked up again:
```
Post order 0x60da8abdbb01b3053c96657a4a94beaf03544d2f2247f46b5da702921dacc542e2f780347649d579452d6dc6960a197e6424580c65c22af8 to OrderBook on chain 11155111
```

You can also see that the distance between two `TRY_AT_EPOCH` messages is exactly five minutes, as expected.
